app/src/main/java/io/github/samolego/canta/MainActivity.kt

```
@@ -72,140 +81,75 @@ class MainActivity : FragmentActivity() {
     }
 
     private fun getPackageInstaller(): PackageInstaller {
         val iPackageInstaller = ShizukuPackageInstallerUtils.getPrivilegedPackageInstaller()
-        val root = Shizuku.getUid() == 0
-        val userId = if (root) android.os.Process.myUserHandle().hashCode() else 0
+        
+        val userId = try {
+            val handle = android.os.Process.myUserHandle()
+            val method = handle.javaClass.getDeclaredMethod("getIdentifier")
+            method.invoke(handle) as Int
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)